macrostep.el

```diff
@@ -602,14 +602,17 @@ behaviors."
                 (macrostep-collapse-overlays-in (point) end)
                 (delete-region (point) end)
                 ;; Create a new overlay
-                (let ((overlay
-                       (make-overlay start
-                                     (if (looking-at "\n")
-                                         (1+ (point))
-                                       (point)))))
+                (let* ((overlay
+                        (make-overlay start
+                                      (if (looking-at "\n")
+                                          (1+ (point))
+                                        (point))))
+                       (back-overlay (unless macrostep-expansion-buffer
+                                       (copy-overlay overlay))))
                   (unless macrostep-expansion-buffer
```

Comment:
That's right. I made your suggested changes as well.

Though I found a bug when giving them a nil priority (when the region surrounds the overlays completely then it didn't work), so now they're given a negative priority.